### ISSUE TITLE
Spike ways to handle "mixed" cop safety

### ISSUE THREAD

**Issue by stdedos** [rubocop/rubocop#13280]
## Is your feature request related to a problem? Please describe.

e.g. the https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation says

> This cop is unsafe in `aggressive` mode, as it cannot be guaranteed that the receiver is actually a string, which can result in a false positive.

But that could mean that, if a receiver could be determined to be a string, then the offence should be auto-correctable safely?

Like in 

```ruby
'Hello' + user.name
user.name + '!!'
Pathname.new('/') + 'test'
```

could become

```ruby
"Hello #{user.name}"
user.name + '!!'
Pathname.new('/') + 'test'
```

even with the cop in `aggressive` mode

## Describe alternatives you've considered

1. There has been a `MixedOffenseSafety` spike PR https://github.com/rubocop/rubocop/pull/13166
2. An "obvious" way could be to downright split cops per-safety (i.e. `Style/StringConcatenation` and `Style/StringConcatenationAggressive`)

## Additional context

_Via https://github.com/rubocop/rubocop/discussions/13278_

**Comment by dvandersluis** [comment]
If `user.name` is `nil`:

```ruby
'Hello' + user.name   #=> TypeError: no implicit conversion of nil into String
"Hello #{user.name}"  #=> "Hello "
```

Therefore the autocorrection is not safe, even if the receiver is a string.

**Comment by Zopolis4** [comment]
Note that there are still other cops where safety can be determined in guaranteed scenarios: `Style/FileEmpty` as in my previous PR, `Lint/BinaryOperatorWithIdenticalOperands` is safe when calling literals, which can be differentiated from methods to determine safety, I think `Lint/OrderedMagicComments` is safe if there is no file encoding magic comment-- there are probably a few more, these are just what I gathered while having a quick glance at Lint cops.
